packages/zero-solid/src/solid-view.ts

```diff
@@ -50,13 +59,21 @@ export class SolidView<V> implements Output {
     onTransactionCommit(this.#onTransactionCommit);
     this.#format = format;
     this.#onDestroy = onDestroy;
+    input.setOutput(this);
+
+    const initialRoot = this.#createEmptyRoot();
+    this.#applyChangesToRoot(
+      input.fetch({}),
+      node => ({type: 'add', node}),
+      initialRoot,
+    );
     [this.#state, this.#setState] = createStore<State>([
-      this.#initialEmptyEntry(),
+      initialRoot,
       queryComplete === true ? complete : unknown,
     ]);
-    input.setOutput(this);
-
-    this.#applyChanges(input.fetch({}), node => ({type: 'add', node}));
+    if (isEmptyRoot(initialRoot)) {
+      this.#builderRoot = this.#createEmptyRoot();
```

Comment:
why not use initialRoot here?

Reply:
To avoid directly modifying the object that we set as the stores state.